.github/workflows/test-freighter-cpp.yml

```diff
@@ -0,0 +1,34 @@
+name: Test Freighter CPP
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'synnax/freighter/cpp/**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
```

Comment:
One of the  useful tools I use in a monorepo is the `dorny/paths-vilter@v2`. It actually does a diff check on the git history as well.